src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeWriter.cs

```diff
@@ -651,5 +656,46 @@ public void Dispose()
             {
             }
         }
+
+        public struct StringLiteralWriter : IDisposable
```

Comment:
I don't see how it would be possible in a single method (except with some particular output parameters). Indeed we don't know how many tokens have to written before the `MaxStringLiteralLength`. A token may be even be split in two string literals.
So I think we have to "start the string literal", write the token one by one until we reach the limit, "close the string", restart a new string literal,...